packages/core/src/contracts/ownedAddressTCRWithAppeals.ts

```diff
@@ -1,17 +1,38 @@
-import { BigNumber } from "bignumber.js";
+import BN from "bignumber.js";
 import { Observable } from "rxjs";
 import "rxjs/add/operator/distinctUntilChanged";
 import "@joincivil/utils";
+
 import { ContentProvider } from "../content/contentprovider";
-import { CivilTransactionReceipt, EthAddress, TxHash, TxData } from "../types";
+import {
+  EthAddress,
+  TwoStepEthTransaction,
+} from "../types";
+import {
+  createTwoStepSimple,
+} from "../utils/contractutils";
 import { requireAccount } from "../utils/errors";
 import { Web3Wrapper } from "../utils/web3wrapper";
 import { BaseWrapper } from "./basewrapper";
 import { OwnedAddressTCRWithAppealsContract } from "./generated/owned_address_t_c_r_with_appeals";
 import { Voting } from "./voting";
-import { EIP20 } from "./eip20";
-import { Parameterizer } from "./parameterizer";
-import { Newsroom } from "./newsroom";
+
+/**
+ * This enum represents the various states a listing can be in.
+ * A listing can be both whitelisted and in the challenge process.
+ */
+export enum ListingState {
+  NOT_FOUND = "NOT FOUND ON REGISTRY",
```

Comment:
Do we care about the values of the enum? I'd say we can leave off the strings and just use auto-incrementing behavior here. Enums are classes and so should be easy to debug without those strings anyways.